hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java

```diff
@@ -357,7 +357,12 @@ public JdbcEnvironmentImpl execute(Connection connection) {
 								);
 							}
 							catch (SQLException e) {
-								log.unableToObtainConnectionMetadata( e );
+								if ( shouldIgnoreMetadataAccessFailure( configurationValues ) ) {
```

Comment:
The test case I added in `MetadataAccessTests` doesn't cover this case, since an actual JDBC connection is required and the `Connection#getMetaData` method needs to throw a `SQLException` to reach it. I didn't really know how to make this happen ☹️

Reply:
I achieved this by mocking the `ConnectionProvider` and the returned `Connection`, but this is pretty dependent on the actual implementation of the `JdbcEnvironmentInitiator`:

```java
@Test
@Jira("https://hibernate.atlassian.net/browse/HHH-18286")
void testDontIgnoreMetadataAccessFailureWhenConnectionCanBeObtained() throws Exception {
	Connection connectionMock = Mockito.mock( Connection.class );
	given( connectionMock.getMetaData() ).willThrow( new SQLException( "Connection closed unexpectedly!" ) );

	ConnectionProvider connectionProviderMock = Mockito.mock( ConnectionProvider.class );
	given( connectionProviderMock.getConnection() ).willReturn( connectionMock );

	StandardServiceRegistryBuilder registryBuilder = new StandardServiceRegistryBuilder();
	registryBuilder.clearSettings();
	registryBuilder.applySetting( JdbcSettings.IGNORE_METADATA_ACCESS_FAILURE_ON_BOOT, false )
			.applySetting( AvailableSettings.CONNECTION_PROVIDER, connectionProviderMock );

	try (StandardServiceRegistry registry = registryBuilder.build()) {
		assertThatExceptionOfType( ServiceException.class )
				.isThrownBy( () -> registry.getService( JdbcEnvironment.class ) )
				.withRootCauseInstanceOf( SQLException.class )
				.havingCause()
				.isInstanceOf( HibernateException.class )
				.withMessage( "Unable to access JDBC metadata" );
	}
}
```

The `QualifiedTableNamingTest` has a `MockedConnectionProvider` implementation of the `ConnectionProvider` interface, which uses the `JdbcMocks` factory class, but that uses dynamic proxies which do not throw a `SQLException` but actually return a result when calling the `Connection#getMetaData` method.